sympy/polys/polyroots.py

```diff
@@ -378,49 +378,57 @@ def roots_binomial(f):
+    # leave it inside a radical (avoid things like cbrt(-x) if possible).
+    coeff, _ = base.as_coeff_Mul()
+    minus = coeff < 0 or base.is_negative
+    if minus:
+        alpha = root(-base, n)
+    else:
+        alpha = root(base, n)
+
+    # It is not generally the case that (a**b)**(1/n) == a**(b/n) however the
+    # difference between these is some factor that is an nth root of unity.
+    # Since we will multiply alpha by all nth roots of unity the full set of
+    # roots generated by either form are equivalent.
+    if alpha.is_Pow and alpha.base.is_Pow and alpha.exp == Rational(1, n):
+        alpha = alpha.base.base ** (alpha.base.exp / n)
 
-    # define some parameters that will allow us to order the roots.
-    # If the domain is ZZ this is guaranteed to return roots sorted
-    # with reals before non-real roots and non-real sorted according
-    # to real part and imaginary part, e.g. -1, 1, -1 + I, 2 - I
-    neg = base.is_negative
     even = n % 2 == 0
-    if neg:
-        if even == True and (base + 1).is_positive:
-            big = True
-        else:
-            big = False
-
-    # get the indices in the right order so the computed
-    # roots will be sorted when the domain is ZZ
-    ks = []
-    imax = n//2
-    if even:
-        ks.append(imax)
-        imax -= 1
-    if not neg:
-        ks.append(0)
-    for i in range(imax, 0, -1):
-        if neg:
-            ks.extend([i, -i])
-        else:
-            ks.extend([-i, i])
-    if neg:
-        ks.append(0)
-        if big:
-            for i in range(0, len(ks), 2):
-                pair = ks[i: i + 2]
-                pair = list(reversed(pair))
-
-    # compute the roots
-    roots, d = [], 2*I*pi/n
-    for k in ks:
-        zeta = exp(k*d).expand(complex=True)
-        roots.append((alpha*zeta).expand(power_base=False))
+
+    if even and not minus:
+        # x**4 = 16, x -> [-2, 2, ...]
+        roots = [-alpha, alpha]
+        for k in range(n // 2 - 1, 0, -1):
+            zeta = exp(2*k*I*pi/n)
+            roots.append(alpha*zeta.conjugate())
+            roots.append(alpha*zeta)
+    elif even and minus:
+        # x**4 = -16, No real roots
+        roots = []
+        for k in range(n // 2 - 1, -1, -1):
+            zeta = exp((2*k + 1)*I*pi/n)
+            roots.append(alpha*zeta.conjugate())
+            roots.append(alpha*zeta)
+    elif not even and not minus:
+        # x**3 = 8, x -> [2, ...]
+        roots = [alpha]
+        for k in range(n // 2, 0, -1):
+            zeta = exp(2*k*I*pi/n)
+            roots.append(alpha*zeta.conjugate())
+            roots.append(alpha*zeta)
+    elif not even and minus:
+        # x**3 = -8, x -> [-2, ...]
+        roots = [-alpha]
+        for k in range(n // 2 - 1, -1, -1):
+            zeta = exp((2*k + 1)*I*pi/n)
+            roots.append(alpha*zeta.conjugate())
+            roots.append(alpha*zeta)
+    else:
+        assert False
```

Comment:
Did you intend to leave this bare assertion? Personally, I like it because you know that all cases are covered and if they (unexpectedly) aren't, you will get the assertion failure. But perhaps something like `assert None  # must be one of above 4 cases` would be more documenting? (Or maybe make a utilities `sanity_check` function to do this in a uniform way when we need it.)

Reply:
I did intend to leave this. I thought it might be more readable if each case above listed its condition explicitly but wanted to make it clear that the 4 cases tested are exhaustive.

Reply:
As you say though a comment could make this more readable.